Update tomcat.adoc

KBbitsP opened the pull request:
adding changes for configuring PGSql database for Kaui

KBbitsP committed:
Update tomcat.adoc

adding changes for configuring PGSql database for Kaui

reshmabidikar commented on userguide/getting-started/includes/tomcat.adoc:
```diff
@@ -177,7 +179,38 @@ In addition to the Kill Bill database, you will also need to configure the Kaui
+
+You should have a database called *kaui* (Or whatever name you have specified in step 1) with the following tables:
 [source,sql]
  kaui_users
```
Good start, but can you create a separate doc called "How to setup Kill Bill/Kaui with PostgresSQL" as discussed in Slack? We do not want to clutter the [Getting Started doc](https://docs.killbill.io/latest/getting_started.html). Once done, can you remove the [Kill Bill Postgres instructions from the development doc](https://docs.killbill.io/latest/development.html#_postgresql_configuration) and point the development doc to this new doc?

KBbitsP replied:
Hi @reshmabidikar yup created another commit, added a separate page for postgreSQL config with 2 sections, one for Killbill and one for Kaui. Pointed both the getting started as well as the development pages to the new one.

reshmabidikar commented on userguide/getting-started/includes/tomcat.adoc:
```diff
@@ -177,7 +179,38 @@ In addition to the Kill Bill database, you will also need to configure the Kaui
+.  Create a database corresponding to kaui. You can run the following command:
+[source,sql]
+ CREATE DATABASE kaui WITH OWNER = killbilluser;
+
```
I think you would also need to specify instructions to create the user called `killbilluser`

KBbitsP replied:
The user will already be created when killbill database is created

reshmabidikar commented on userguide/getting-started/includes/tomcat.adoc:
```diff
@@ -177,7 +179,38 @@ In addition to the Kill Bill database, you will also need to configure the Kaui
+.  Modify the `TOMCAT_HOME\webapps2\ROOT\WEB-INF\config\database.yml` adapter on line 3 as :
+[source,properties]
+  <% adapter = java.lang.System.getProperty('kaui.db.adapter', ENV['KAUI_DB_ADAPTER'] || 'postgresql') %>
+
```
Can this be specified within the `TOMCAT_HOME/conf/catalina.properties` ?

KBbitsP replied:
No, this is specifically to be done on database.yml

reshmabidikar replied:
@pierre, I may be wrong, but I think specifying this in the `catalina.properties` should also work since we specify system properties here too?

reshmabidikar replied:
FYI, I was able to get postgres to work by setting `kaui.db.adapter=postgresql` in `catalina.properties`. Something you might want to mention in the doc.

reshmabidikar commented on userguide/getting-started/includes/tomcat.adoc:
```diff
@@ -177,7 +179,38 @@ In addition to the Kill Bill database, you will also need to configure the Kaui
+.  Grant privileges to the schema using the following command:
+[source,sql]
+  GRANT ALL ON SCHEMA public TO killbilluser;
+
```
These steps look different from the Kill Bill Postgres steps [here](https://docs.killbill.io/latest/development.html#_postgresql_configuration). Could you verify both sets of instructions once?

KBbitsP replied:
Yes, initially I granted only on the tables but there was errors in the logs regarding sequences. So I granted on all to the user.

KBbitsP committed:
Added PostgreSQL config page

Added a separate page for PostgreSQL configuration and pointed the getting started and Development guides to the new page

reshmabidikar commented on userguide/getting-started/includes/tomcat.adoc:
```diff
@@ -52,7 +52,7 @@ As mentioned above, you need to configure the Kill Bill database manually. For t
 
 
-. If you are using PostgreSQL, you can refer to the detailed instructions for PostgreSQL setup in our development document https://docs.killbill.io/latest/development.html#_postgresql_configuration[here] 
+. If you are using PostgreSQL, you can refer to the detailed instructions for PostgreSQL setup in our development document https://docs.killbill.io/latest/PostgreSQL.html#_postgresql_configuration[here] 
```
```suggestion
. If you are using PostgreSQL, you can refer to the detailed instructions for PostgreSQL setup in our PostgresSQL configuration document https://docs.killbill.io/latest/PostgreSQL.html#_postgresql_configuration[here] 
```

reshmabidikar commented on userguide/getting-started/includes/tomcat.adoc:
```diff
@@ -177,7 +179,11 @@ In addition to the Kill Bill database, you will also need to configure the Kaui
 
-. You should have a database called *kaui* (Or whatever name you have specified in step 1) with the following tables:
+
+. If you are using PostgreSQL, you can refer to the detailed instructions for PostgreSQL setup in our development document https://docs.killbill.io/latest/PostgreSQL.html#_postgresql_kaui_configuration[here] 
```
```suggestion
. If you are using PostgreSQL, you can refer to the detailed instructions for PostgreSQL setup in our PostgreSQL configuration document https://docs.killbill.io/latest/PostgreSQL.html#_postgresql_kaui_configuration[here] 
```

reshmabidikar commented on userguide/tutorials/PostgreSQL.adoc:
```diff
@@ -0,0 +1,66 @@
+==== PostgreSQL Configuration
+
```
Check heading levels, the top-level should be H2

reshmabidikar commented on userguide/tutorials/PostgreSQL.adoc:
```diff
@@ -0,0 +1,66 @@
+==== PostgreSQL Configuration
+
+In order to configure PostgreSQL, you need to follow the steps given below (These steps can be executed either via the command line *psql* tool or via *pgAdmin*):
```
Also, adding a small introductory paragraph might help

reshmabidikar commented on userguide/tutorials/PostgreSQL.adoc:
```diff
@@ -0,0 +1,66 @@
+[source,sql]
+  GRANT ALL ON SCHEMA public TO killbilluser;
+
+.  Modify the `PROJECT_ROOT/profiles/killbill/src/main/resources/killbill-server.properties` OR `TOMCAT_HOME\conf\catalina.properties` as the case may be. Add the following properties :
```
```suggestion
.  Modify the `PROJECT_ROOT/profiles/killbill/src/main/resources/killbill-server.properties` OR `TOMCAT_HOME/conf/catalina.properties` as the case may be. Add the following properties :
```

reshmabidikar commented on userguide/tutorials/development.adoc:
```diff
@@ -182,41 +182,7 @@ In order to configure MySQL, you need to follow the steps given below (These ste
-. Grant privileges to the schema using the following command:
-[source,sql]
-  GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA killbillschema TO killbilluser;
+. If you are using PostgreSQL, you can refer to the detailed instructions for PostgreSQL setup in our development document https://docs.killbill.io/latest/PostgreSQL.html#_postgresql_configuration[here] 
```
```suggestion
. If you are using PostgreSQL, you can refer to the detailed instructions for PostgreSQL setup in our PostgresSQL configuration document https://docs.killbill.io/latest/PostgreSQL.html#_postgresql_configuration[here] 
```

reshmabidikar commented on userguide/tutorials/PostgreSQL.adoc:
```diff
@@ -0,0 +1,66 @@
+[source,sql]
+ CREATE DATABASE kaui WITH OWNER = killbilluser;
+
+.  Connect to the kaui database created above. In psql, you can run the following:
```
Do we need to create a separate schema for `kaui` like done for `killbill`?

reshmabidikar commented on userguide/tutorials/PostgreSQL.adoc:
```diff
@@ -0,0 +1,66 @@
+  kaui.db.url=jdbc:postgresql://127.0.0.1:5432/kaui
+  kaui.db.username=killbilluser
+  kaui.db.password=killbill
+
```
Could you add a similar section to the Kill Bill PostgreSQL configuration section above just for the sake of consistency?

KBbitsP committed:
Merge branch 'killbill:v3' into v3

KBbitsP committed:
Merge branch 'killbill:v3' into v3

KBbitsP committed:
Kaui Schema and Sequences grant

Added separate schema for Kaui as well as added separate grant for sequences

KBbitsP committed:
Merge branch 'v3' of https://github.com/KBbitsP/killbill-docs into v3

pierre commented on userguide/tutorials/PostgreSQL.adoc:
```diff
@@ -0,0 +1,78 @@
+
+.  Modify the `TOMCAT_HOME/webapps2/ROOT/WEB-INF/config/database.yml` adapter on line 3 as :
+[source,properties]
+  <% adapter = java.lang.System.getProperty('kaui.db.adapter', ENV['KAUI_DB_ADAPTER'] || 'postgresql') %>
```
Instead of modifying the line, can't the user simply set `KAUI_DB_ADAPTER= postgresql` (e.g. in Tomcat setenv.sh)?

The default line is the following, so the environment variable `KAUI_DB_ADAPTER` should be read by default:

```
<% adapter = java.lang.System.getProperty('kaui.db.adapter', ENV['KAUI_DB_ADAPTER'] || 'mysql2') %>
```

reshmabidikar replied:
@pierre, do you mean [this](https://github.com/killbill/killbill-docs/pull/370#discussion_r895414417) ?

pierre replied:
Ah, yes, that works too and no need for the user to tweak the yaml file (which could be a bit error prone). Sorry I missed it, too many GitHub notifications 🙈

KBbitsP committed:
Merge branch 'killbill:v3' into v3

KBbitsP committed:
Added the DB adapter property

Added the DB adapter property to be added to catalina.properties

KBbitsP reviewed (commented):
Hi @reshmabidikar added the suggested adapter property

pierre commented:
Thanks!
